### PR TITLE
Adding ktor-test-server app for auth poc

### PR DESCRIPTION
**Pull request by ZacharyRizer**

### CODE REVIEW

File: spikes/ktor-test-server/src/main/kotlin/dextest/plugins/Auth.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+)
+
+// Dummy appConfig, replace with actual config loading logic
+val authConfig = AuthConfig(
+    issuerUrl = "https://apigw-stg.cdc.gov:8443",
+    introspectionUrl = "https://your-introspection-url",
+    requiredScopes = "dex:upload"
+)
+
+fun Application.configureAuth() {
```
Comment (mkrystof) [inline spikes/ktor-test-server/src/main/kotlin/dextest/plugins/Auth.kt:36]: Please test and make sure this plays nicely with the graphQL module. The graphQL module has its own way of doing auth so my only concern at this point is that these two will integrate successfully.

Comment (ZacharyRizer) [reply spikes/ktor-test-server/src/main/kotlin/dextest/plugins/Auth.kt:36]: @mkrystof are you referring the current `Auth0` package setup in the `Graphql.kt` file? The part that looks like this: 

`    if (securityEnabled) {
        install(Authentication) {
            jwt {
                jwt("auth-jwt") {
                    realm = myRealm
                    verifier(
                        JWT
                            .require(Algorithm.HMAC256(Base64.getDecoder().decode(secret)))
                            .withAudience(audience)
                            .withIssuer(issuer)
                            .build()
                    )
                    validate { credential ->
                        if (credential.payload.getClaim("username").asString() != "") {
                            JWTPrincipal(credential.payload)
                        } else {
                            logger.error("username missing from JWT claims, denying the token")
                            null
                        }
                    }
`

**Review by mkrystof — COMMENTED** [review]
This looks great, Zach!  I see the unit tests, which look fantastic.  However, do you have any postman or any other supporting external test scaffolding you can check in as part of this PR?

**Comment by ZacharyRizer** [comment]
> This looks great, Zach! I see the unit tests, which look fantastic. However, do you have any postman or any other supporting external test scaffolding you can check in as part of this PR?

Unfortunately no. Most of my testing was manual with Postman and relied on the Keycloak credentials. This PR doesn't have any protected/wrapped routes in it. But when I was testing out the middleware, I wrapped a fake `test-route` to handle the testing. I took out that test route when I finalized this PR.

**Review by mkrystof — APPROVED** [review]

**Pull request merged**